tests/core/full_node/test_block_height_map.py

```diff
@@ -183,6 +197,15 @@ async def test_restore_entire_chain(self, tmp_dir, db_version):
 
             height_map = await BlockHeightMap.create(tmp_dir, db_wrapper)
 
+            # We replaced the whole cache at this point so all values should be different
+            async with aiofiles.open(tmp_dir / "height-to-hash", "rb") as f:
+                new_heights = bytearray(await f.read())
+                # Make sure we wrote the whole file
+                assert len(new_heights) == (10000 + 1) * 32
+                # Make sure none of the old values remain
+                for i in range(0, len(heights), 32):
+                    assert new_heights[i : i + 32] != heights[i : i + 32]
+
```

Comment:
It's different because in this test we initially set this up with 900 items and then expose that to 10000 DB items that are completely different. We're restoring the entire chain from DB, rendering the whole cache file invalid.